optuna/study.py

```diff
@@ -653,22 +834,14 @@ def _pop_waiting_trial_id(self) -> Optional[int]:
         return None
 
     def _ask(self) -> trial_module.Trial:
-        # Sync storage once at the beginning of the objective evaluation.
-        self._storage.read_trials_from_remote_storage(self._study_id)
-
-        trial_id = self._pop_waiting_trial_id()
-        if trial_id is None:
-            trial_id = self._storage.create_new_trial(self._study_id)
-        return trial_module.Trial(self, trial_id)
+        # TODO(hvy): Remove this method since `Study.ask` has been implemented.
```

Comment:
I left these as TODOs since I know that we have users of those private previous ask and tell APIs, such as the Hydra Optuna sweeper. Cc. @toshihikoyanase . However, the logic is changed so it might still break. Maybe we can remove them at this point.